docs/user_guide/datasets/acs_income.rst

```diff
@@ -0,0 +1,165 @@
+.. _acsincome-data:
+Retiring Adult: ACSIncome
+-------------------------
+
+Introduction
+^^^^^^^^^^^^^^^^^
+
+The ACSIncome dataset is one of five datasets created by Ding et al. [#0]_ 
+as an improved alternative to the popular UCI Adult dataset.
```

Comment:
Are you planning on adding a description for UCI Adult as well? Just wondering because this leaves it unclear what the improvements actually are, and what the problems with UCI Adult were in the first place. It could totally be a separate page, of course. You can absolutely borrow heavily from the paper (with citation). No need to reinvent the wheel 😉

Reply:
I wasn't planning to, at least not in this PR. To be honest, I was a bit confused what the improvements were. It's definitely more data (1.6M rows vs. ~40k), and it's more recent data (2018 vs. 1994). The paper questions the choice of $50k income cutoff, which is why we're providing the raw incomes here. I'd need to do more analysis to understand how the composition of these datasets differ. Let's discuss more as a team about what we think a good dataset for fairness benchmarking means

Reply:
If we do end up adding more information to this page about issues with the UCI Adult dataset (and I agree that even a few paragraphs would provide useful context), it might be good to move the information about the unique values in each variable to an appendix/the end of the page and just include the field names + description in this table; otherwise, the length of the table might be unwieldy.

Reply:
One of the reasons I'm asking: UCI Adult documentation might be the foundation for documentation of these ACS datasets, and perhaps on a separate page.

Re: limitations: I definitely agree on dataset size and recency as well as the choice of cutoff. Others from the paper:
> The choice of the 50,000 dollar threshold is idiosyncratic and potentially limits the external validity of UCI Adult as a benchmark for algorithmic fairness. In 1994, the median USincome was 26,000 dollars, and 50,000 dollars corresponds to the 76th quantile of the income distribution, and the 88th and 89th quantiles of the income distribution for the Black and female populations, respectively. Consequently, almost all of the Black and female instances in the dataset fall below the threshold and models trained on UCI adult tend to have substantially higher accuracies on these subpopulations. For instance, a standard logistic regression model trained on UCI Adult dataset achieves 85% accuracy overall, 91.4% accuracy on the Black instances, and 92.7% on Female instances. This is a rather untypical situation since often machine learning models perform more poorly on historically disadvantaged groups.
The following chart also tells you how much influence this choice of threshold has on the fairness objectives that we're trying to optimize for: 
![image](https://user-images.githubusercontent.com/10245648/149057359-7320374a-5ed7-47ec-a2c1-eea47e8a7d97.png)

More: 
> A notable if obvious limitation of our work is that it is entirely US-centric. A richer dataset ecosystem covering international contexts within the algorithmic fairness community is still lacking. Although empirical work in the Global South is central in other disciplines, there continues to be much need for the North American fairness community to engage with it more strongly

Another limitation IMO is the task definition that doesn't strike me as realistic. After all, why would you want to predict people's income? There's probably a different motivation there. That's probably a larger point of discussion, though. The main ones I wanted to include are the ones mentioned in the paper. I agree that this may be better off in a separate PR, though.

@alliesaizan what do you mean by "unique values in each variable"? Did you mean the feature descriptions? I personally don't feel very strongly about including in the text or at the end, but it's certainly something we can consider as more content is added.

Reply:
@romanlutz my apologies if I was not clear! I mean putting the part of the feature description that reviews what values the variable can take at the end, but agree that we can reevaluate this suggestion when more content has been added.

Reply:
Agreed with @romanlutz that even a sentence or two of context for what the UCI Adult dataset is and why an improved alternative might be needed will be helpful for readers who may not have that context!

Reply:
I added a brief summary of UCI Adult and the improvements offered by ACSIncome based on the points made in this thread :)